ext/random/engine_mt19937.c

```diff
@@ -357,19 +357,19 @@ PHP_METHOD(Random_Engine_Mt19937, __unserialize)
 	/* members */
 	t = zend_hash_index_find(d, 0);
 	if (!t || Z_TYPE_P(t) != IS_ARRAY) {
-		zend_throw_exception(NULL, "Incomplete or ill-formed serialization data", 0);
+		zend_throw_error(NULL, "Invalid serialization data for Random\\Engine object");
```

Comment:
Can anyone tell me how I can insert the actual class name here? The `__unserialize` method is `@implementation-alias`.

Reply:
`engine->std.ce->name` should contain the class name.